MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1913,6 +1915,153 @@ SeparateImagesInMemoryMap (
   return;
 }
 
+/**
+  Split memory map descriptors based on the input special region list. After the function, one or more memory map
+  descriptors will divide evenly into every special region so attributes can be targetted at those regions. Every
+  descriptor covered by a special region will have its virtual address set to SPECIAL_REGION_PATTERN.
+
+  @param[in, out] MemoryMapSize                   IN:   The size, in bytes, of the old memory map before the split
+                                                  OUT:  The size, in bytes, of the used descriptors of the split
+                                                        memory map
+  @param[in, out] MemoryMap                       IN:   A pointer to the buffer containing a sorted memory map
+                                                  OUT:  A pointer to the updated memory map
+  @param[in]      DescriptorSize                  The size, in bytes, of an individual EFI_MEMORY_DESCRIPTOR
+  @param[in]      ExpandedMemMapSize              The size, in bytes, of the full memory map buffer

```

Comment:
The name of this variable changed.  Should be BufferSize.

Reply:
Removed